src/ValueCalculator.java

```diff
@@ -0,0 +1,67 @@
+import java.util.Arrays;
+
+public class ValueCalculator {
+
+    private double[] values;
+    private int size;
+    private int halfSize;
+    private double[] firstHalf;
+    private double[] secondHalf;
+    private long start;
+    private long end;
```

Comment:
These are supposed to be the local method variables. You do not need them here at the class level